plugin/al/lib/AL_USDMaya/AL/usdmaya/nodes/Scope.cpp

```diff
@@ -0,0 +1,296 @@
+  MPlug(thisMObject(), MPxTransform::scalePivotTranslate).setLocked(true);
+  MPlug(thisMObject(), MPxTransform::scalePivot).setLocked(true);
+  MPlug(thisMObject(), MPxTransform::rotatePivotTranslate).setLocked(true);
+  MPlug(thisMObject(), MPxTransform::rotatePivot).setLocked(true);
+  MPlug(thisMObject(), MPxTransform::shearXY).setLocked(true);
+  MPlug(thisMObject(), MPxTransform::shearXZ).setLocked(true);
+  MPlug(thisMObject(), MPxTransform::shearYZ).setLocked(true);
+}
+
+//----------------------------------------------------------------------------------------------------------------------
+Scope::Scope()
+{
+}
+
+//----------------------------------------------------------------------------------------------------------------------
+Scope::~Scope()
+{
+}
+
+//----------------------------------------------------------------------------------------------------------------------
+MPxTransformationMatrix* Scope::createTransformationMatrix()
+{
+  TF_DEBUG(ALUSDMAYA_EVALUATION).Msg("Scope::createTransformationMatrix\n");
+  return new BasicTransformationMatrix;
+}
+
+//----------------------------------------------------------------------------------------------------------------------
+MStatus Scope::initialise()
+{
+  TF_DEBUG(ALUSDMAYA_EVALUATION).Msg("Scope::initialise\n");
+  const char* const errorString = "Scope::initialise";
+  try
+  {
+    setNodeType(kTypeName);
+
+    addFrame("USD Prim Information");
+    m_primPath = addStringAttr("primPath", "pp", kReadable | kWritable | kStorable | kConnectable | kAffectsWorldSpace, true);
+    m_inStageData = addDataAttr("inStageData", "isd", MayaUsdStageData::mayaTypeId, kWritable | kStorable | kConnectable | kHidden | kAffectsWorldSpace);
+
+    mustCallValidateAndSet(m_primPath);
+    mustCallValidateAndSet(m_inStageData);
+
+    for(auto& inAttr : {m_primPath, m_inStageData})
+    {
+      AL_MAYA_CHECK_ERROR(attributeAffects(inAttr, translate), errorString);
+      AL_MAYA_CHECK_ERROR(attributeAffects(inAttr, rotate), errorString);
+      AL_MAYA_CHECK_ERROR(attributeAffects(inAttr, rotateOrder), errorString);
+      AL_MAYA_CHECK_ERROR(attributeAffects(inAttr, scale), errorString);
+      AL_MAYA_CHECK_ERROR(attributeAffects(inAttr, shear), errorString);
+      AL_MAYA_CHECK_ERROR(attributeAffects(inAttr, rotatePivot), errorString);
+      AL_MAYA_CHECK_ERROR(attributeAffects(inAttr, rotatePivotTranslate), errorString);
+      AL_MAYA_CHECK_ERROR(attributeAffects(inAttr, scalePivot), errorString);
+      AL_MAYA_CHECK_ERROR(attributeAffects(inAttr, scalePivotTranslate), errorString);
+      // Maya 2018 (checked 2018.2 and 2018.3) has a bug where, if any loaded plugin has an MPxTransform subclass
+      // that has ANY attribute that connected to rotateAxis, it will cause the rotateAxis to evaluate INCORRECTLY,
+      // even on the BASE transform class!
+      // See this gist for full reproduction details:
+      //   https://gist.github.com/elrond79/f9ddb277da3eab2948d27ddb1f84aba0
```

Comment:
Do you happen to know if we have an internal bug for this?  If not we should get one logged.  It does sound familiar.

Reply:
Not sure...(If that was addressed to me)